source/adapters/hip/common.cpp

```diff
@@ -164,13 +164,19 @@ void detail::ur::assertion(bool Condition, const char *pMessage) {
 
 // Global variables for UR_RESULT_ADAPTER_SPECIFIC_ERROR
 thread_local ur_result_t ErrorMessageCode = UR_RESULT_SUCCESS;
-thread_local char ErrorMessage[MaxMessageSize];
+thread_local char ErrorMessage[MaxMessageSize]{};
 
 // Utility function for setting a message and warning
 [[maybe_unused]] void setErrorMessage(const char *pMessage,
                                       ur_result_t ErrorCode) {
   assert(strlen(pMessage) < MaxMessageSize);
+  // Copy MaxMessageSize - 1 bytes to ensure the resultant string is always null
```

Comment:
Maybe "copy *at most* MaxMessageSize - 1" bytes?

Reply:
At least, that's how I infer `strncpy_s` to work. I see that `strncpy` seems to copy all MaxMessageSize - 1 characters after all.